### PR TITLE
update metric to new metric spec

### LINKED ISSUE #355: Update metric YAML for dbt-core 1.6+
### Describe the bug

The spec of the YAML for dbt_metrics and metricflow is completely different so dbt-core 1.6+ is breaking on compile (as we have a metric defined in the integration tests for this package). 

We should update this metric definition to use the new syntax.

### PR DESCRIPTION
**Pull request by dave-connors-3**
This is a:
- [x] bug fix PR with no breaking changes
- [ ] new functionality

## Link to Issue 
Closes #355

## Description & motivation

This PR updates the unpack logic for the new metric spec, and updates the integration test project accordingly to match the spec. Even though we had initially decided to remove metric references altogether, this option at least gives us the ability to add metric checks in the future.  Happy to remove metrics altogether if we still feel like that's a better move. 

This PR will not pass CI until all adapters have a 1.6 release. 

## Checklist
- [ ] I have verified that these changes work locally on the following warehouses (Note: it's okay if you do not have access to all warehouses, this helps us understand what has been covered)
    - [ ] BigQuery
    - [x] Postgres
    - [ ] Redshift
    - [ ] Snowflake
    - [ ] Databricks
    - [ ] DuckDB
- [ ] I have updated the README.md (if applicable)
- [ ] I have added tests & descriptions to my models (and macros if applicable)

### CODE REVIEW

**Comment by b-per** [comment]
Thanks Dave! 

Let's see how we can get it passing at least some part of CI before merging 😄 (I don't think that we'll get the DuckDB one due to the 1.6 restriction)

**Comment by graciegoheen** [comment]
> The semantic layer requires a 'metricflow_time_spine' model in the project, but none was found. Guidance on creating this model can be found on our docs site (https://docs.getdbt.com/docs/build/metricflow-time-spine) 

@dave-connors-3 can you add this date spine to the PR?

**Comment by dave-connors-3** [comment]
@graciegoheen done!

an important note: adding this model to the integration tests project required adding exceptions to the root models and naming convention rule checks -- I decided to add these as global execeptions in the `dbt_project_evaluator_exceptions.csv` that ships with the package, since this model is virtually guaranteed to be a root model that doesn't adhere to your naming conventions for all users. Let me know if this feels like a bad call!

**Comment by dave-connors-3** [comment]
@graciegoheen @b-per this _should_ be good to go. I was able to test on all but databricks (🤞 ) and the two failures we have here are a result of the two non-dbt-Labs adapters not yet cutting a release >=1.6.0rc1. we're at their mercy to do so for this to pass without an update to our CI logic!

File: integration_tests/models/marts/metrics.yml [hunk 1]
```diff
@@ -1,29 +1,37 @@
...
-metrics:
-  - name: new_customers
-    label: New Customers marked 'paying'
-    model: ref('dim_model_7')
-    description: "The number of paid customers using the product"
 
-    calculation_method: count
-    expression: user_id # superfluous here, but shown as an example
+semantic_models:
+  - name: my_model # A semantic model with the name Transactions
```
Comment (graciegoheen) [inline integration_tests/models/marts/metrics.yml:5]: Is the name Transactions? or my_model?

File: integration_tests/models/marts/metrics.yml [hunk 2]
```diff
@@ -1,29 +1,37 @@
...
+    model: ref('dim_model_7') # References the dbt model named `fact_transactions`
```
Comment (graciegoheen) [inline integration_tests/models/marts/metrics.yml:6]: > References the dbt model named `fact_transactions`

?
it looks like it references dbt_model_7

File: models/marts/structure/fct_model_naming_conventions.sql [hunk 3]
```diff
@@ -4,6 +4,7 @@
 with all_graph_resources as (
     select * from {{ ref('int_all_graph_resources') }}
     where not is_excluded
+    and resource_name != 'metricflow_time_spine'
```
Comment (graciegoheen) [inline models/marts/structure/fct_model_naming_conventions.sql:7]: Can you add a comment here as well?

File: integration_tests/models/marts/metrics.yml (hunk 1 above)
Comment (dave-connors-3) [inline integration_tests/models/marts/metrics.yml:5]:
```suggestion
  - name: my_model 
```

File: integration_tests/models/marts/metrics.yml (hunk 2 above)
Comment (dave-connors-3) [reply integration_tests/models/marts/metrics.yml:6]: both just copy paste from docs, will remove

Comment (dave-connors-3) [inline integration_tests/models/marts/metrics.yml:6]:
```suggestion
    model: ref('dim_model_7') 
```

File: models/marts/structure/fct_model_naming_conventions.sql [hunk 4]
```diff
@@ -4,6 +4,7 @@
 with all_graph_resources as (
     select * from {{ ref('int_all_graph_resources') }}
     where not is_excluded
```
Comment (dave-connors-3) [inline models/marts/structure/fct_model_naming_conventions.sql:6]:
```suggestion
    where not is_excluded
    -- exclude required metricflow time spine
```

File: models/marts/.DS_Store
Comment (graciegoheen) [inline models/marts/.DS_Store:1]: Can we delete this and add .DS_Store to the .gitignore file?

File: integration_tests/models/marts/exposures.yml [hunk 5]
```diff
@@ -13,6 +13,7 @@ exposures:
     
     depends_on:
       - ref('fct_model_6')
+      - ref('fct_model_9')
```
Comment (graciegoheen) [inline integration_tests/models/marts/exposures.yml:16]: Why was this added?

Comment (dave-connors-3) [reply integration_tests/models/marts/exposures.yml:16]: i don't remember! will remove and see if i break anything

Comment (dave-connors-3) [reply integration_tests/models/marts/exposures.yml:16]: ah, I had to materialize `dim_model_7` for the metrics I made to work, so I added this here to make sure we still had an ephemeral model feeding the exposure for this check

File: models/marts/.DS_Store
Comment (dave-connors-3) [reply models/marts/.DS_Store:1]: I added it to the .gitignore, but I don't see this in my local filetree so I don't know how to delete! how strange

**Review by Jstein77 — APPROVED** [review]
LGTM!

**Review by b-per — APPROVED** [review]
